stdlib/domain.ml

```diff
@@ -19,7 +19,21 @@
 module Raw = struct
   (* Low-level primitives provided by the runtime *)
   type t = private int
-  external spawn : (unit -> unit) -> Mutex.t -> t
+
+  (* The layouts of [state] and [term_sync] are hard-coded in
+     [runtime/domain.c] *)
+
+  type 'a state =
+    | Running
+    | Finished of ('a, exn) result
+
+  type 'a term_sync = {
+    mutable state : 'a state [@warning "-69"] ; (* protected by [mut] *)
```

Comment:
I am not sure why I get this warning either. Note that it complains about the `mutable` part.

This looks like a false positive to me as `term_sync` is an input to an `external` which can very well (and does) mutate `state`.

Reply:
I wonder if we should make the code more regular by changing `spawn` to return the result directly. Changing it to return an `('a, exn) result` value would be very simple, but we could go further by returning an `'a` and raising the corresponding exceptions -- I think there would be opportunities for simplifying the code there.

```ocaml
      match
        DLS.create_dls ();
        DLS.set_initial_keys pk;
        let res = f () in
        res
      with
      | res ->
        do_at_exit ();
        res
      | exception exn ->
        let bt = Printexc.get_raw_backtrace () in
        (try do_at_exit () with _ -> ());
        Printexc.raise_with_backtrace bt exn
```

Reply:
Ah, I didn't know that the warning was about the lack of mutation in the code.

I agree that the warning is a false positive and that it is fine to silence it. (Maybe a comment that explains what warning arises and why we silence it could be nice.)